### PR TITLE
feat(front end): link compare charts page to backend

### PR DESCRIPTION
**Pull request by gargi-bansal**
### Purpose

Linked compare charts page to the backend

[HRT-69](https://softeng761team5.atlassian.net/browse/HRT-69)

### Approach
The front end makes API calls to grab all selected charts from the back end.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR involve a UI change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/pages/CompareCharts.tsx [hunk 1]
```diff
@@ -1,64 +1,143 @@
...
+  const fetchCharts = async (): Promise<void> => {
+    await fetch(`${API_DOMAIN}course/${params.courseID}`, {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${cookies["accessToken"]}`,
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+    })
+      .then((res) => res.json())
```
Comment (lucycy98) [inline src/pages/CompareCharts.tsx]: can we please add some code here such that if res.status is not 200, then we send them to the dashboard? to keep it consistent, look in DisplayCards (function: fetchDimensions).

File: src/pages/CompareCharts.tsx [hunk 2]
```diff
@@ -1,64 +1,143 @@
...
+      .then((data) => {
+        setCourseName(createCourse(data[0]).name);
+      });
+    let chartsArray: Chart[] = [];
+
+    await Promise.all(
+      Object.keys(chartsToCompareIDs).map(async (key: string) => {
```
Comment (lucycy98) [inline src/pages/CompareCharts.tsx:63]: sorry can we also do a quick check if chartsToCompareIDs is null or undefined? then we could redirect to dashboard again or deal with it someway else. thanks!

**Review by lucycy98 — DISMISSED** [review]
left some quick comments to do with error checking, but other than that looks good :)

**Review by TCHE614 — CHANGES_REQUESTED** [review]
If a fully incomplete chart is compared with a completed chart nothing gets displayed. Another issue is when a fully incomplete chart is compared with 2 completed chart the fully incomplete chart becomes either 0 or 100 for the marks depending on the selection. 

image

image
the chart from 10:38pm 11/10 has is not completed so it technically shouldn't be there

I think it might just be easier to disable compare charts if an incomplete chart is chosen. 
Other than that looks good to me good job guys!

**Comment by gargi-bansal** [comment]
> If a fully incomplete chart is compared with a completed chart nothing gets displayed. Another issue is when a fully incomplete chart is compared with 2 completed chart the fully incomplete chart becomes either 0 or 100 for the marks depending on the selection.
> 
> image
> 
> image
> the chart from 10:38pm 11/10 has is not completed so it technically shouldn't be there
> 
> I think it might just be easier to disable compare charts if an incomplete chart is chosen.
> Other than that looks good to me good job guys!

@TCHE614 made your requested changes!

**Review by mitchfaulcon — CHANGES_REQUESTED** [review]
This should be a super easy fix but can you change it so the back button goes to the correct course page instead of the dashboard?

**Review by TCHE614 — DISMISSED** [review]

**Comment by gargi-bansal** [comment]
> This should be a super easy fix but can you change it so the back button goes to the correct course page instead of the dashboard?

@mitchfaulcon fixed it!

**Review by mitchfaulcon — APPROVED** [review]

**Review by preetpatel — APPROVED** [review]

**Pull request merged**